### PR TITLE
Remove All Fireproof Sites Feature

### PR DESCRIPTION
**Pull request by nalcalag**
Task/Issue URL: [https://app.asana.com/0/72649045549333/1201953975647701/f](https://app.asana.com/0/72649045549333/1201953975647701/f)

### Description
New toolbar menu option for removing all fireproof sites

### Steps to test this PR

_Remove All Fireproof sites_
- Install from this branch
- Go to Settings > Fireproof Sites and make sure you have 1 or more fireproof sites added
- Tap on toolbar overflow menu > Remove All
- [x] Check all Fireproof sites are removed
- [x] Check empty state text appears

_Undo Remove All Fireproof sites_
- Do all previous steps from _Remove all Fireproof sites_
- [x] Check Snackbar shows with `undo` option
- Tap on "undo" option
- [x] Check all Fireproof sites are back in the list

### UI changes
| Before  | After |
| ------ | ----- |
Screenshot_20220427-204034_DuckDuckGo | Screenshot_20220427_203855
|

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/fire/fireproofwebsite/ui/FireproofWebsitesViewModel.kt [hunk 1]
```diff
@@ -59,6 +60,7 @@ class FireproofWebsitesViewModel @Inject constructor(
 
     private val fireproofWebsites: LiveData<List<FireproofWebsiteEntity>> = fireproofWebsiteRepository.getFireproofWebsites()
     private val fireproofWebsitesObserver = Observer<List<FireproofWebsiteEntity>> { onPreservedCookiesEntitiesChanged(it!!) }
+    private var removedWebsitesEntities: List<FireproofWebsiteEntity> = emptyList()
```
Comment (joshliebe) [inline app/src/main/java/com/duckduckgo/app/fire/fireproofwebsite/ui/FireproofWebsitesViewModel.kt:63]: Could we remove keeping this state by passing the list along with the command, then just pass these back to `ViewModel` in the undo action in the `Snackbar`?

Something like: https://github.com/duckduckgo/Android/blob/b5dbb62bd910ecb76a207fe31371b9d1224783a7/app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt#L326

Comment (joshliebe) [inline app/src/main/java/com/duckduckgo/app/fire/fireproofwebsite/ui/FireproofWebsitesViewModel.kt:63]: Could we remove keeping this state by passing the list along with the command, then just pass these back to `ViewModel` in the undo action in the `Snackbar`?

Something like: https://github.com/duckduckgo/Android/blob/b5dbb62bd910ecb76a207fe31371b9d1224783a7/app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt#L326

Comment (nalcalag) [reply app/src/main/java/com/duckduckgo/app/fire/fireproofwebsite/ui/FireproofWebsitesViewModel.kt:63]: Interesting approach, let me check 👍

**Review by joshliebe — APPROVED** [review]
Works great, good job! 🎉

**Pull request merged**